packages/next/shared/lib/router/utils/prepare-destination.ts

```diff
@@ -121,6 +121,7 @@ export function prepareDestination(args: {
   const query = Object.assign({}, args.query)
   delete query.__nextLocale
   delete query.__nextDefaultLocale
+  delete query._nextDataReq
```

Comment:
Should we add another underscore to match the other internal parameters?

Reply:
Sounds good, updated in https://github.com/vercel/next.js/pull/37574/commits/c65c67f2628f52374163128f182e45222a42aaa8